### PR TITLE
Config path

### PR DESCRIPTION
**Pull request by molant**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/sonarwhal/sonarwhal)
- [x] Followed the [commit message guidelines](https://sonarwhal.com/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- [x] Added/Updated related documentation.
- [x] Added/Updated related tests.

## Short description of the change(s)

- Add support for paths in the configuration
- Rename `getRuleName` to `getRulePath`

### CODE REVIEW

**Comment by molant** [comment]
This is finally ready to review 🎉

**Comment by molant** [comment]
@alrra seems to be a platform issue with a couple of the tests. Looking into it.

**Comment by molant** [comment]
@alrra, @sarvaje this should be (hopefully) ready to review.

File: packages/rule-no-vulnerable-javascript-libraries/tests/tests.ts [hunk 1]
```diff
@@ -62,6 +62,6 @@ const userMediumConfigTests: Array<RuleTest> = [
     }
 ];
 
-ruleRunner.testRule(getRuleName(__dirname), defaultTests);
-ruleRunner.testRule(getRuleName(__dirname), userHighConfigTests, { ruleOptions: { severity: 'high' } });
-ruleRunner.testRule(getRuleName(__dirname), userMediumConfigTests, { ruleOptions: { severity: 'medium' } });
+ruleRunner.testRule(getRulePath(__filename), defaultTests);
+ruleRunner.testRule(getRulePath(__filename), userHighConfigTests, { ruleOptions: { severity: 'high' } });
+ruleRunner.testRule(getRulePath(__filename), userMediumConfigTests, { ruleOptions: { severity: 'medium' } });
```
Comment (alrra) [inline packages/rule-no-vulnerable-javascript-libraries/tests/tests.ts]: Put this in a variable?

**Comment by molant** [comment]
@alrra @sarvaje feedback addressed and conflict solved 🎉

File: packages/sonarwhal/src/lib/config.ts [hunk 2]
```diff
@@ -262,6 +261,79 @@ export class SonarwhalConfig {
...
+
+        const stat = fs.statSync(configRoot); //eslint-disable-line
+        const configPath = stat.isDirectory() ? configRoot : path.dirname(configRoot);
+
+        if (!configPath) {
+            return config;
+        }
+
+        const resolve = (value: string): string => {
+            if (!value.startsWith('.')) {
```
Comment (molant) [inline packages/sonarwhal/src/lib/config.ts]: I'll explain this a bit more.

File: packages/sonarwhal/src/lib/config.ts [hunk 3]
```diff
@@ -262,6 +261,93 @@ export class SonarwhalConfig {
...
+         */
+        const stat = fs.statSync(configRoot); //eslint-disable-line
+        const configPath = stat.isDirectory() ? configRoot : path.dirname(configRoot);
+
+        if (!configPath) {
+            return config;
+        }
+
+        /**
+         * If `value` is a relative path (ie, it starts with `.`), it transforms it
```
Comment (alrra) [inline packages/sonarwhal/src/lib/config.ts]: > i.e.

**Comment by alrra** [comment]
Merged as https://github.com/sonarwhal/sonarwhal/commit/1e2e54f8c9a46a1d11df219c8c7f2ac6b33c1a72 & https://github.com/sonarwhal/sonarwhal/commit/232564d4e80b163208b0295faacaa98e53a07da4 .

**Pull request closed without merging**